FileDescriptor.cpp

```diff
@@ -178,7 +177,7 @@ static void checkCanonicalBaseName(const char *path) {
         to<std::string>("checkCanonicalBaseName(", path, "): (", name,
                         ") doesn't match canonical base (", base, ")"));
   }
-#else
+else
```

Comment:
this causes the __APPLE__/macOS build to fail:

https://travis-ci.org/facebook/watchman/jobs/417265100#L537

Was this an intentional change?

It's not super clear to me how this fixes the freebsd build, because the conditional